test/hotspot/gtest/utilities/test_ostream.cpp

```diff
@@ -50,6 +51,32 @@ static void test_stringStream_is_zero_terminated(const stringStream* ss) {
   ASSERT_EQ(ss->base()[ss->size()], '\0');
 }
 
+static size_t count_char(const char* s, size_t len, char ch) {
+  size_t cnt = 0;
+
+  for (size_t i = 0; i < len; ++i) {
+    if (s[i] == ch) {
+      cnt++;
+    }
+  }
+  return cnt;
+}
+
+static size_t count_char(const stringStream* ss, char ch) {
+  return count_char(ss->as_string(), ss->size(), ch);
```

Comment:
I don't think STL is allowed. 

Make sense. ss->as_string() is not necessary. 

I don't like the idea we assume ss is always zero-terminated like C-string. There is a member variable _written in class stringStream.  Technically speaking, the implementation can avoid from writing '\0' in the end.  that's why I would like to use len argument.  For me, `count_char(ss->base(), ss->size(), ch)` is more reliable because it depends on interfaces instead of implementation.  an interface is supposed to be more stable than implementation.